### DISCUSSION TITLE
Compile for ESP32 s2/s3

### DISCUSSION THREAD

**Discussion by 203Null** [hathach/tinyusb#1112] [Q&A] [answered]
I tried to compile for ESP32 S2/S3 and I'm currently stuck on this

> `idf.py -B_build/espressif_addax_1 -DFAMILY=esp32s3 -DBOARD=espressif_addax_1  clean
process_begin: CreateProcess(NULL, env python C:\esp-idf\tools\idf.py -B_build/espressif_addax_1 -DFAMILY=esp32s3 -DBOARD=espressif_addax_1 clean, ...) failed.
make (e=2): The system cannot find the file specified.
make: *** [clean] Error 2`

Anyone knows how can I properly set up ESP IDF for this?

**Comment by kkitayam** [comment]
That error message looks like same as https://github.com/hathach/tinyusb/discussions/1030#discussioncomment-1200414
Maybe it's because python.exe is not in PATH environment variable. We can check whether python.exe exists or not on PATH environment variable by followings

```console
C:\>where python.exe
C:\Program Files\Python39\python.exe
```

**Reply by 203Null** [reply]
![image](https://user-images.githubusercontent.com/19933093/135020342-3dbb1c45-9859-4418-90ee-f71456040be1.png)

**Reply by 203Null** [reply]
![image](https://user-images.githubusercontent.com/19933093/135020457-bf467169-06ce-4f6c-a16c-d31eb05f0584.png)
I think it's fine. I can access idf.py perfectly fine and all idf examples compiles fine

**Reply by 203Null** [reply]
I can compile for STM32F1/STM32F4 with arm-none-eabi-gcc just fine

**Comment by 203Null** [comment]
I made some progress after hardcode my python and idf.py in but I still can't get it to compile 
![image](https://user-images.githubusercontent.com/19933093/135043264-be3c1717-eb42-49bc-ae91-19065cb74a5e.png)

**Comment by hathach** [comment]
which example you are testing with, the IDF use freeRTOS therefore only example ends with `_freertos` is supported. Otherwise, you need to do some example portings.

**Reply by 203Null** [reply]
I will give it a try}, thx

**Reply by 203Null** [reply]
I got it to build for ESP32-S2 but not S3

ESP32-S2:
![image](https://user-images.githubusercontent.com/19933093/135121219-5823a78a-eb54-4653-9794-6e1119a2884f.png)

ESP32-S3:
![image](https://user-images.githubusercontent.com/19933093/135121418-fc38f8e3-0d4b-40f9-8820-510d4bc97380.png)

However this is not a tinyUSB issue, espressif straight up didn't give ESP32-S3 usb driver 
![image](https://user-images.githubusercontent.com/19933093/135121768-916ff4c6-86ca-44d8-91e3-9e508474e310.png)
No soc/usb_periph.c or soc/usb_periph.h found.

**Comment by 203Null** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/19933093/135141646-38743014-2536-4130-95f3-9895de74b68e.png)
I got it to work on the esp32-s3-devkitc-1 with the latest development branch of ESP-IDF. I will post a compile guide later.

**Reply by Huckies** [reply]
I'm encountering the same problem, would you please post a guide?
